File: java/FizzBuzz.java
import java.util.stream.IntStream;

public class FizzBuzz {

    public static void main(String[] args) {
        fizzBuzz(100);
    }
    
    public static void fizzBuzz(int end) {
        IntStream.rangeClosed(1, end)
                .mapToObj(i -> i % 15 == 0 ? "FizzBuzz" : i % 3 == 0 ? "Fizz" : i % 5 == 0 ? "Buzz" : i)
                .forEach(s -> System.out.println(s));
    }
}

File: elixir/fizzbuzz.exs
defmodule FizzBuzz do
  def fizzbuzz(last) do
     Stream.iterate(1, &(&1 + 1))
     |> Stream.map(fn i when rem(i, 15) == 0 -> :FizzBuzz
                      i when rem(i,  3) == 0 -> :Fizz
                      i when rem(i,  5) == 0 -> :Buzz
                      i -> i end)
     |> Enum.take(last)
     |> Enum.each(&IO.puts/1)
  end
end

FizzBuzz.fizzbuzz(100)
